swan-lake/integration-tutorials/sending-a-message-to-a-service.md

```diff
@@ -35,56 +35,82 @@ Follow the instructions given in this section to develop the service.
-
-    function initializeHttpClient() returns http:Client|error => new (healthcareBackend);
+    ```json
+    {
+        "name": "thomas collins",
+        "hospital": "grand oak community hospital",
+        "category": "surgery",
+        "availability": "9.00 a.m - 11.00 a.m",
+        "fee": 7000
+    }
     ```
 
-    The argument to the `new` expression is the URL for the backend service.
+    ![Paste JSON as record](/learn/images/tutorial_sending_a_message_to_a_service_paste_json_as_record.gif)
+    
+    1. Copy the JSON.
+    2. Open VS [Code Command palette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette).
+    3. Search and select `Ballerina: Paste JSON as record` command.
+    
+    > **Note:** You can manually define the record, if there is no sample JSON.
+    > 
+    > ```ballerina
+    > type Doctor record {|
+    >     string name;
+    >     string hospital;
+    >     string category;
+    >     string availability;
+    >     decimal fee;
+    > |};
+    > ```
 
-5. Define a record corresponding to the payload from the backend service.
+    The payload will be an array of JSON objects in which, the structure of each JSON object matches this record.
 
-    ```ballerina
-    type Doctor record {|
-        string name;
-        string hospital;
-        string category;
-        string availability;
-        decimal fee;
-    |};
-    ```
+5. Define the [HTTP service (REST API)](https://ballerina.io/learn/by-example/#rest-service) that has the resource that accepts user requests, retrieves relevant details from the backend service, and responds to the request.
 
-    The payload will be an array of JSON objects in which, the structure of each JSON object matches this record. Note that you can use the "Paste JSON as record" VS Code command to generate the record if you have the JSON payload.
+    - Use `/healthcare` as the service path (or the context) of the service, which is attached to the listener listening on port `port`.
 
-6. Define the [HTTP service (REST API)](https://ballerina.io/learn/by-example/#rest-service) that has the resource that accepts user requests, retrieves relevant details from the backend service, and responds to the request. Use `/healthcare` as the service path (or the context) of the service, which is attached to the listener listening on port `port`. Define an HTTP resource that allows the `GET` operation on resource path `/querydoctor` and accepts the `category` (corresponding to the specialization) as a path parameter.
+        ![Define the service](/learn/images/tutorial_sending_a_message_to_a_service_define_a_service.gif)
 
-    ```ballerina
-    service /healthcare on new http:Listener(port) {
-        resource function get querydoctor/[string category]() {
-            
+    - Define an HTTP resource that allows the `GET` operation on resource path `/querydoctor` and accepts the `category` (corresponding to the specialization) as a path parameter.
```

Comment:
IMO, `querydoctor` is not a proper resource name. Best practice is to use nouns instead of verbs for resource function names. So having `doctors` should be enough IMO.

Refer to best practice: https://learn-ballerina.github.io/best_practices/naming_conventions.html#rest-service-and-resource-paths

@MaryamZi wdyt?

Reply:
+1. We've used the same as the WSO2 tutorial, but I agree it needs to change.

Reply:
+1 for this.